### PR TITLE
[ci-skip] Upgrade deployment from 1.4.x to 2.2.x

### PR DESCRIPTION
**Pull request by Mohitv**
Signed-off-by: Mohitv <mohit_vaish@hotmail.com>

**Changelog**
- Add Upgrade deployment of Hypeledger Fabric deployment to 2.2.x. The documentation is available in docs folder at /docs/source/operations/upgrade_fabricnetwork_1.4.x_2.2.x.md
 

**Reviewed by**
@sownak @jagpreetsinghsasan

 

**Linked issue**
#1845

### CODE REVIEW

File: platforms/hyperledger-fabric/configuration/roles/upgrade/orderer-capabilities/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+# Create the build directory
+- name: Create build directory if it does not exist
+  file:
+    path: "{{ build_path }}/channel-artifacts/"
+    state: directory
+
+# Create a capabilities json
+- name: Create capabilities json
+  copy:
+    content: "{{ '{\"channel\":{\"mod_policy\":\"Admins\",\"value\":{\"capabilities\":{\"V2_0\":{}}},\"version\":\"0\"},\"orderer\":{\"mod_policy\":\"Admins\",\"value\":{\"capabilities\":{\"V2_0\":{}}},\"version\":\"0\"},\"application\":{\"mod_policy\":\"Admins\",\"value\":{\"capabilities\":{\"V2_0\":{}}},\"version\":\"0\"}}' | from_json | to_nice_json}}"
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/upgrade/orderer-capabilities/tasks/main.yaml:26]: Same comment as above

File: platforms/hyperledger-fabric/configuration/roles/upgrade/peer-binary/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# This role creates value file for each orderer
+- name: Create value file for each orderer in org
+  include_tasks: create_peer_value.yaml
+  loop: "{{ component_services.peers }}"
+  loop_control:
+    loop_var: peer
+  when:
+    - component_services.peers is defined
+
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/upgrade/peer-binary/tasks/main.yaml:31]: Please remove any extra lines (most probably this above line has newline and some spaces) and just add a newline at EOF

**Comment by jagpreetsinghsasan** [comment]
Please squash merge the commits into 1 and re-request for review.
Until you hit the re-request review button, the reviewers will not be sure if you are done doing the changes or not.
Here's how to do it (you can see the re-request review button on the right side of the screenshot)
image

File: docs/source/operations/upgrade_fabricnetwork_1.4.x_2.2.x.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+<a name = "upgrading-fabric"></a>
+# Upgrading Hyperledger Fabric version from 1.4.x to 2.2.x
+
+- [Pre-requisites](#pre_req)
+- [Steps to upgrade](#upgrade_steps)
+
+<a name = "pre_req"></a>
+## Pre-requisites
+A DLT system which is using Hyperledger Fabric version 1.4.x and is setup using Bevel automation framework
```
[the remark this answers is not in the document]
Comment (Mohitv) [reply docs/source/operations/upgrade_fabricnetwork_1.4.x_2.2.x.md]: Done

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM

**Comment by jagpreetsinghsasan** [comment]
@Mohitv the PR looks perfectly fine to me. Just one last comment, to squash merge the commits into one.

**Comment by suvajit-sarkar** [comment]
The code is test ok, closing this issue with a bug create for a minor fix #2142

**Pull request merged**